pkg/nfd-client/topology-updater/nfd-topology-updater.go

```diff
@@ -88,7 +88,7 @@ func (w *nfdTopologyUpdater) Run() error {
 
 	podResClient, err := podres.GetPodResClient(w.resourcemonitorArgs.PodResourceSocketPath)
 	if err != nil {
-		klog.Fatalf("failed to get PodResource Client: %w", err)
+		klog.Errorf("failed to get PodResource Client: %w", err)
```

Comment:
Sorry for noticing this a bit late, but I took a closer look at `cmd/nfd-topology-updater/main.go` and we do a `klog.Exit(err)` on the return value there. Thus, I suggest to change like this
 
```suggestion
		return fmt.Errorf("failed to get PodResource Client: %w", err)
```

(applies to the other changes you made all in `Run()`, too)